### DISCUSSION TITLE
Navegação sem fazer novas requests

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [frontendbr/forum#1660] [Perguntas e Respostas] [answered]
Fala galera! Bom dia!

Minha dúvida é: em meu site atualmente feito em react quando o cara entra na tela "home" faço uma requisição com axios para os dados da home, quando ele navega para a tela "sobre"  o axios faz a request dos dados dela... Mas quando o cara volta atualmente faço a request dos dados da "home" novamente, não queria isso... quero que ao voltar já traga os dados que já foram carregados antes.

1 - Posso fazer isto com redux, sempre que carrego os dados coloco nele, e ao voltar a tela home só uso aqueles dados... isso é uma boa prática? Sabem se o próprio Axios ou o Sagas já tem algo que faça isso por mim?

Meu medo é o browser do cara ficar lento por causa deste estado que vai ter um tanto de informações de cache além das informações do estado da app mesmo.

Gostaria de opiniões,
muito obrigado, abraços!

**Comment by kivervinicius** [comment, marked as answer]
usa o https://github.com/RasCarlito/axios-cache-adapter ou o https://github.com/kuitos/axios-extensions os dois vão resolver seu problema, usei por mto tempo o extensions porém agora migrei para o cache-adapter devido a ter que trabalhar offline, só isso já resolve seu problema sem ter que recorrer a soluções mirabolantes sem necessidade.

O adapter pode ser trabalhado só em memória ou via indexedb (localforage) o extensions somente via memoria, o bom de trabalhar nem que seja com sessionStorage que mesmo se o usuário atualizar a página a solicitação está la. Claro que há tempo de expiração de acordo com seu caso de uso.

**Reply by bistocchi** [reply]
Opa kiver, poderia me passar seu contato para ajudar ?

**Reply by kivervinicius** [reply]
@bistocchi https://www.linkedin.com/in/kiverteixeira/ desculpa a demora
